### DISCUSSION TITLE
Allow so_keepalive as valid option for listen

### DISCUSSION THREAD

**Discussion by PackiarajSakkananGW** [Kong/kong#9158] [Ideas and feature requests]
aws NLB has fixed idle connection timeout which is 350s, the workaround is to send tcp keepalive. more information is available [here](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html#connection-idle-timeout)

Ngnix has built in support for keep-alive on its listen socket, more information is available [here](http://nginx.org/en/docs/http/ngx_http_core_module.html#listen)  

For unclear reasons Kong allows only selected set of listen configuration, refer [here](https://github.com/Kong/kong/blob/master/kong/conf_loader/listeners.lua#L17) for supported options 

accept so_keepalive as valid option so that NLB timeout can be controlled.

BTW: istio provides a workaround for this https://github.com/istio/istio/issues/28879

**Comment by windmgc** [comment]
Modify the nginx configuration template used by Kong is also allowed, please follow the instructions here https://docs.konghq.com/gateway/2.8.x/reference/configuration/#custom-nginx-templates--embedding-kong

In your scenario you may need to add `so_keepalive` option here https://github.com/Kong/kong/blob/a0c3296938eed5acb71bc533204fab136718da21/kong/templates/nginx_kong.lua#L78
